neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -34,9 +35,23 @@ public ProtocolHandler(NeoSystem system)
             this.sentHashes = new FIFOSet<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2);
         }
 
+        private void FlushHashes()
+        {
+            knownHashes.Clear();
+            sentHashes.Clear();
+        }
+
         protected override void OnReceive(object message)
         {
-            if (!(message is Message msg)) return;
+            if (!(message is Message msg))
```

Comment:
I think that this can easily be a source for attacks.

Maybe, for now, these flushes can just happen on neo-cli, no?

Reply:
Already discussing this above with @erikzhang